include/tvm/relay/error_reporter.h

```diff
@@ -0,0 +1,104 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file error_reporter.h
+ * \brief The set of errors raised by Relay.
+ */
+#ifndef TVM_RELAY_ERROR_REPORTER_H_
+#define TVM_RELAY_ERROR_REPORTER_H_
+
+#include <tvm/relay/error.h>
+#include <tvm/relay/expr.h>
+#include <string>
+#include <sstream>
+#include <vector>
+
+namespace tvm {
+namespace relay {
```

Comment:
can we put this into error.h? assuming this is part of error handling mechanism

Reply:
There was a cyclic dependency issue that I was trying to avoid by splitting the files, a few places use `relay::Error` which `ErrorReporter` depends on.